opm/grid/cpgrid/CpGridData.cpp

```diff
@@ -512,380 +1199,393 @@ void createInterfaces(std::vector<std::map<int,char> >& attributes,
     }
-    else
-    {
-        // Force update of the sync counter in the remote indices.
-        cell_remote_indices_.getModifier<false,false>(0);
-    }
+    return map2Local;
+}
 
-    // We can identify existing cells with the help of the index set.
-    // Now we need to compute the existing faces and points. Either exist
-    // if they are reachable from an existing cell.
-    // We use std::numeric_limits<int>::max() to indicate non-existent entities.
-    std::vector<int> face_indicator(view_data.geometry_.geomVector<1>().size(),
-                                    std::numeric_limits<int>::max());
-    std::vector<int> point_indicator(view_data.geometry_.geomVector<3>().size(),
-                                     std::numeric_limits<int>::max());
-    for(ParallelIndexSet::iterator i=cell_indexset_.begin(), end=cell_indexset_.end();
-            i!=end; ++i)
+std::vector<std::set<int> > computeAdditionalFacePoints(const std::vector<std::array<int,8> >& globalCell2Points,
+                                                        const OrientedEntityTable<0, 1>& globalCell2Faces,
+                                                        const Opm::SparseTable<int>& globalFace2Points,
+                                                        const GlobalIdSet& globalIds)
+{
+    std::vector<std::set<int> > additionalFacePoints(globalCell2Points.size());
+
+    for ( std::size_t c = 0; c < globalCell2Points.size(); ++c)
     {
-        typedef boost::iterator_range<const EntityRep<1>*>::iterator RowIter;
-        int row_index=i->global();
-        // Somehow g++-4.4 does not find functions of father even if we
-        // change inheritance of OrientedEntityTable to public.
-        // Therfore we use an ugly cast to base class here.
-        const Opm::SparseTable<EntityRep<1> >& c2f=view_data.cell_to_face_;
-        for(RowIter f=c2f[row_index].begin(), fend=c2f[row_index].end();
-            f!=fend; ++f)
-        {
-            int findex=f->index();
-            --face_indicator[findex];
-            // points reachable from a cell exist, too.
-            for(auto p=view_data.face_to_point_[findex].begin(),
-                    pend=view_data.face_to_point_[findex].end(); p!=pend; ++p)
+        const auto& points = globalCell2Points[c];
+        for(const auto& face: globalCell2Faces[EntityRep<0>(c, true)])
+            for(const auto& point: globalFace2Points[face.index()])
             {
-                assert(*p >= 0);
-                --point_indicator[*p];
+                auto candidate = std::find(points.begin(), points.end(), point);
+                if(candidate == points.end())
+                    // point is not a corner of the cell
+                    additionalFacePoints[c].insert(globalIds.id(EntityRep<3>(point,true)));
             }
-        }
     }
+    return additionalFacePoints;
+}
 
+template<bool send, class Map2Global, class Map2Local>
+void createInterfaceList(
+#if DUNE_VERSION_NEWER(DUNE_GRID, 2, 7)
```

Comment:
a lot of these. can they be centralized and type aliased?

Reply:
Go idea and I totally agree. Will do that. Will also be less error prone (did find a few typos in some of these.